### PR TITLE
thud: rename from go-thumbnailer, 0.1.0 -> 0.3.0

### PR DESCRIPTION
**Pull request by donovanglover**
## Description of changes

Changelog: https://github.com/donovanglover/thud/blob/0.3.0/CHANGELOG.md
Release: https://github.com/donovanglover/thud/releases
Changes: https://github.com/donovanglover/thud/compare/0.1.0...0.3.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by donovanglover** [comment]
@emilylange Since you merged the original `go-thumbnailer`, do you mind helping me with this PR?

**Comment by donovanglover** [comment]
Result of `nixpkgs-review pr 249224` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>thud</li>
  </ul>
</details>

File: pkgs/by-name/th/thud/package.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+  cargoHash = "sha256-Hk3HlcA253FAA9hw5p9W+Mvec84zLo7bEmM2/BbmjiM=";
+
+  nativeBuildInputs = [
+    installShellFiles
+  ];
+
+  postInstall = ''
+    mkdir -p $out/share/thumbnailers
+    substituteAll ${./thud.thumbnailer} $out/share/thumbnailers/thud.thumbnailer
```
Comment (emilylange) [inline pkgs/by-name/th/thud/package.nix:26]: I don't think we need to vendor `thud.thumbnailer`.

How about

```suggestion
    mkdir -p $out/share/thumbnailers
    substitute assets/thud.thumbnailer $out/share/thumbnailers/thud.thumbnailer \
      --replace "thud" "$out/bin/thud"
```

or

```suggestion
    install -Dm644 assets/thud.thumbnailer $out/share/thumbnailers/thud.thumbnailer
    substituteInPlace $out/share/thumbnailers/thud.thumbnailer --replace "thud" "$out/bin/thud"
```

**Review by emilylange — CHANGES_REQUESTED** [review]
Looks good!

(just one nitpick)

File: pkgs/by-name/th/thud/package.nix (hunk 1 above)
Comment (donovanglover) [reply pkgs/by-name/th/thud/package.nix]: Fixed

**Review by emilylange — APPROVED** [review]

**Pull request merged**